no-undefined-types when use infer after 54.1.0

<!--
NOTE 1: It would help if you can first scan the issue tracker before submitting
a new issue: https://github.com/gajus/eslint-plugin-jsdoc/issues

NOTE 2: If you are running ESLint/`eslint-plugin-jsdoc` from within an IDE,
before filing an issue, please be sure to restart the IDE after making any
updates to ensure you are indeed getting errors with the updated version.
Running `eslint` on the command line may also be helpful in allowing you
to better provide us with the error details including the stack trace.

NOTE 3: If you have a bug specifically related to the parsing of types, e.g.,
the `some-unrecognized+type-syntax` (within curly brackets) in:

/**
 * @param {some-unrecognized+type-syntax} myName
 */

...then please file instead at https://github.com/simonseyock/jsdoc-type-pratt-parser/issues/
-->

<!--
TIP: If you have a complicated config with `overrides`, you can simplify the
config you provide us by running `eslint` with `--print-config` and the file
that is triggering the error. See:
https://eslint.org/docs/user-guide/command-line-interface#print-config
-->

## Expected behavior
<!-- Provide a detailed description of how you expected the software to -->
<!-- behave. -->
No errors in console when use infer like before 54.1.0

## Actual behavior
<!-- Provide a detailed description of how the software actually behaved, -->
<!-- including any rationale for why that behavior is incorrect. -->
After 54.1.0 this code shows error
% eslint .

/myfile.js
  8:1  error  The type 'I' is undefined  jsdoc/no-undefined-types
  8:1  error  The type 'I' is undefined  jsdoc/no-undefined-types

## ESLint Config

<!--

What is the minimal config that reproduces the issue?

Please try to reproduce with only a single rule in your config (along
with the minimal number of options or settings that are needed to trigger
the error).

-->
```js
import jsdoc from "eslint-plugin-jsdoc";
import globals from "globals";

export default [
	{
		plugins: { jsdoc },

		languageOptions: {
			globals: {
				...globals.node,
				NodeJS: true,
			},

			sourceType: "commonjs",
		},

		rules: {
			"jsdoc/no-undefined-types": ["error", { markVariablesAsUsed: false }],
		},
	},
];

```

## ESLint sample

<!-- What code triggers the error? -->

```js
// Format JS code here
/**
 * @template T
 */
class Test {}

/**
 * @template {Test} T
 * @typedef {T extends Test<infer I> ? I : never} TestType
 */

console.log(Test);
```

<!-- OR, if this is not triggered by a single rule, what is the
   problem and how can one reproduce? -->

## Environment

- Node version: <!-- $ node -v -->v25.6.0
- ESLint version <!-- $ npx eslint -v -->v9.39.2
- `eslint-plugin-jsdoc` version: 62.5.3

--- a/docs/rules/no-undefined-types.md
+++ b/docs/rules/no-undefined-types.md
@@ -1240,5 +1240,16 @@ class Test {}
  * @template {Test} T
  * @typedef {T extends Test<infer I> ? I : never} TestType
  */
+
+/**
+ * @template T
+ */
+class Test {
+  /** @typedef {T} Type */
+  /** @type {Type} */
+  t;
+}
+
+console.log(Test);
 ````
 

--- a/src/rules/noUndefinedTypes.js
+++ b/src/rules/noUndefinedTypes.js
@@ -235,9 +235,15 @@ export default iterateJsdoc(({
     ancestorNodes.flatMap((ancestorNode) => {
       return getTemplateTags(ancestorNode);
     }) :
-    utils.getPresentTags([
-      'template',
-    ]);
+    // We err on the side of being too aggressive; checking only
+    //   present tags is not sufficient
+    comments.flatMap((doc) => {
+      return doc.tags.filter(({
+        tag,
+      }) => {
+        return tag === 'template';
+      });
+    });
 
   const closureGenericTypes = templateTags.flatMap((tag) => {
     return utils.parseClosureTemplateTag(tag);

--- a/test/rules/assertions/noUndefinedTypes.js
+++ b/test/rules/assertions/noUndefinedTypes.js
@@ -2137,5 +2137,19 @@ export default /** @type {import('../index.js').TestCases} */ ({
          */
       `,
     },
+    {
+      code: `
+        /**
+         * @template T
+         */
+        class Test {
+          /** @typedef {T} Type */
+          /** @type {Type} */
+          t;
+        }
+
+        console.log(Test);
+      `,
+    },
   ],
 });

